Models/Visualization/Graph.cs

```diff
@@ -0,0 +1,460 @@
+                    {
+                        dataGraph.AddEdge(new DataEdge(vertex, v) { Text = "\"FAIL\"" });
+                    });
+                }
+                else if (x is ConditionalInteraction)
+                {
+                    var x2 = (ConditionalInteraction)x;
+                    getTriggerOrEventVertex(vertexDict, x2.finishedTrigger, v =>
+                    {
+                        dataGraph.AddEdge(new DataEdge(vertex, v) { Text = "\"FINISHED\"" });
+                    });
+                    if (x2.triggerList != null)
+                    {
+                        foreach (var t in x2.triggerList)
+                        {
+                            getTriggerOrEventVertex(vertexDict, t, v =>
+                            {
+                                dataGraph.AddEdge(new DataEdge(v, vertex) { Text = "\"CONDITION\"" });
+                            });
+                        }
+                    }
+                }
+                else if (x is PersistentTokenInteraction)
+                {
+                    var x2 = (PersistentTokenInteraction)x;
+                    getTriggerOrEventVertex(vertexDict, x2.eventToActivate, v =>
+                    {
+                        dataGraph.AddEdge(new DataEdge(vertex, v) { Text = "\"ACTIVATE\"" });
+                    });
+                    getTriggerOrEventVertex(vertexDict, x2.alternativeTextTrigger, v =>
+                    {
+                        dataGraph.AddEdge(new DataEdge(vertex, v) { Text = "\"ALT. TEXT\"" });
+                    });
+                }
+                else if (x is ThreatInteraction)
+                {
+                    var x2 = (ThreatInteraction)x;
+                    getTriggerOrEventVertex(vertexDict, x2.triggerDefeatedName, v =>
+                    {
+                        dataGraph.AddEdge(new DataEdge(vertex, v) { Text = "\"DEFEATED\"" });
+                    });
+                }
+                else if (x is TextInteraction)
+                {
+                    // No extra triggers
+                }
+
+                // Special case: StoryPointInteractionsa re only used during procedural generation for debug purposes
+                else if (x is Procedural.StoryElements.StoryPointInteraction)
+                {
+                    var x2 = (Procedural.StoryElements.StoryPointInteraction)x;
+
+                    // Endpoints are simple
+                    foreach (var otherAfter in x2.OtherAfterTriggers)
+                    {
+                        getTriggerOrEventVertex(vertexDict, otherAfter, v =>
+                        {
+                            dataGraph.AddEdge(new DataEdge(vertex, v) { Text = "\"ENDING\"" });
+                        });
+                    }
+                }
+                else
+                {
+                    throw new NotImplementedException("Graph preparation not implemented for interaction type: " + x.GetType().Name);
```

Comment:
Let's change this to a Debug.Log statement or something and not throw an Exception, because for files that already have an event like that, it crashes the program and doesn't allow them to continue. e.g. I had a file with a MultiEventInteraction in it.